File: 00-力扣/02-中等题/03-盛水最多的容器.go
package main
func maxArea(height []int) int {
	L :=0
	R :=len(height)-1
	m :=0
 for L<R {
 cur := min(height[L],height[R]) * (R-L)
 if cur > m {
		 m = cur
 }
 if height[L] < height [R] {
		 L++
 }else {
		 R--
 }

 }
 return m
}
func min(m int,n int) int {
 if m > n {
		 return n
 }else {
		 return m
 }
}
func main()  {
	var height = []int{1,1}
	maxArea(height)
}

File: 00-力扣/02-中等题/03-盛水最多的容器.js
var maxArea = function (height) {
  let left = 0;
  let right = height.length - 1;
  let res = 0;
  while (left < right) {
    min = Math.min(height[left], height[right]);
    let area = min * (right - left);
    res = Math.max(area, res);
    // if (height[left] <= height[right]) {
    //   ++left;
    // } else {
    //   --right;
    // }
    height[left] <= height[right] ? ++left : right--;
  }
  return res;
};
